XiEditor/StyleMap.swift

```diff
@@ -14,13 +14,87 @@
 
 import Cocoa
 
+extension NSFont {
+    /// If the font is monospace, returns the width of a character, else returns 0.
+    func characterWidth() -> CGFloat {
+        if self.isFixedPitch {
+            let characters = [UniChar(0x20)]
+            var glyphs = [CGGlyph(0)]
+            if CTFontGetGlyphsForCharacters(self, characters, &glyphs, 1) {
+                let advance = CTFontGetAdvancesForGlyphs(self, .horizontal, glyphs, nil, 1)
+                return CGFloat(advance)
+            }
+        }
+        return 0
+    }
+}
+
 /// A represents a given text style.
 struct Style {
-    var fgColor: NSColor;
-    var bgColor: NSColor;
-    var weight: UInt16;
-    var underline: Bool;
-    var italic: Bool;
+    var font: NSFont
+    var fgColor: NSColor
+    var bgColor: NSColor
+    var setTextBgColor: Bool
```

Comment:
Maybe my thinking is somehow influenced, because I've created a Sublime color scheme recently (which is not to be confused with a sublime ui theme). There you have a default foreground and background color, but setting the default text background color gives you

<img width="645" alt="screen shot 2017-05-09 at 07 12 22" src="https://cloud.githubusercontent.com/assets/409021/25836635/c7a41d76-3488-11e7-801f-d2ab01595734.png">

and not

<img width="645" alt="screen shot 2017-05-09 at 07 01 51" src="https://cloud.githubusercontent.com/assets/409021/25836638/cb18a274-3488-11e7-992d-a8758a9fb315.png">


The same applies to the gutter foreground and background color (which are part of the color scheme in Sublime).

<img width="645" alt="screen shot 2017-05-09 at 07 15 15" src="https://cloud.githubusercontent.com/assets/409021/25836640/cf9f8402-3488-11e7-9dd4-aae8bbd650de.png">

vs

<img width="645" alt="screen shot 2017-05-09 at 07 15 24" src="https://cloud.githubusercontent.com/assets/409021/25836643/d27ed7fe-3488-11e7-9e95-79b43251402e.png">

Which I think is the expected behaviour when creating a color scheme and I wouldn't consider both examples a part of a distinct UI-theming.

This example is also where the `setTextBgColor` comes from. Because when I am setting the "global" textview/gutter background, it is not necessary to also set the text background (even though you wouldn't notice because they are of the same color - but it felt wrong).

Again, I am open to the change these things, just want to give my reasoning first.